flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/assigners/MySqlSnapshotSplitAssigner.java

```diff
@@ -180,6 +189,28 @@ private void captureNewlyAddedTables() {
         }
     }
 
+    private void asynchronouslySplitIfNeed() {
+        if (remainingTables.size() > 0) {
+            if (executor == null) {
+                ThreadFactory threadFactory =
+                        new ThreadFactoryBuilder().setNameFormat("snapshot-split").build();
+                this.executor = Executors.newSingleThreadExecutor(threadFactory);
+            }
+
+            executor.submit(
+                    () -> {
+                        while (remainingTables.size() > 0) {
+                            TableId nextTable = remainingTables.pollFirst();
```

Comment:
How we make sure the table splits generation has finished if failover happened ?

Reply:
Thank you for your comments. I have revised the code again.